packages/core/admin/admin/src/components/NpsSurvey/index.js

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+
+import { Flex, Button, Typography } from '@strapi/design-system';
+import { usePersistentState } from '@strapi/helper-plugin';
+
+const delays = {
+  postResponse: 90 * 24 * 60 * 60 * 1000, // 90 days in ms
+  postFirstDismissal: 7 * 24 * 60 * 60 * 1000, // 7 days in ms
+  postSubsequentDismissal: 90 * 24 * 60 * 60 * 1000, // 90 days in ms
+};
+
+const checkIfShouldShowSurvey = (settings) => {
+  const { enabled, lastResponseDate, firstDismissalDate, lastDismissalDate } = settings;
+
+  if (!enabled) {
+    return false;
+  }
+
+  if (lastResponseDate) {
+    // If the user has already responded to the survey
+    const timeSinceLastResponse = Date.now() - new Date(lastResponseDate).getTime();
+
+    if (timeSinceLastResponse >= delays.postResponse) {
+      return true;
+    }
+
+    return false;
+  }
+
+  if (lastDismissalDate) {
+    // If the user has dismissed the survey twice or more before
+    const timeSinceLastDismissal = Date.now() - new Date(lastDismissalDate).getTime();
+
+    if (timeSinceLastDismissal >= delays.postSubsequentDismissal) {
+      return true;
+    }
+
+    return false;
+  }
+
+  if (firstDismissalDate) {
+    // If the user has dismissed the survey before
+    const timeSinceFirstDismissal = Date.now() - new Date(firstDismissalDate).getTime();
+
+    if (timeSinceFirstDismissal >= delays.postFirstDismissal) {
+      return true;
+    }
+
+    return false;
+  }
+
+  // If the user has not interacted with the survey before
+  return true;
+};
+
+export function useNpsSurveySettings() {
+  const [npsSurveySettings, setNpsSurveySettings] = usePersistentState('NPS_SURVEY_SETTINGS', {
+    enabled: true,
+    lastResponseDate: null,
+    firstDismissalDate: null,
+    lastDismissalDate: null,
+  });
+
+  return { npsSurveySettings, setNpsSurveySettings };
+}
```

Comment:
Also here could benefit why you're exporting it, because i didn't realise at first but it looks like the ambition is we can set it to `enabled` in other parts of the application?

Reply:
Yes it's only used in the register form, because we set enabled to true or false depending on whether the user checked the subscribe to newsletter checkbox. I don't imagine it being used elsewhere in the future.

I could also store the object in the Register form by calling localStorage directly. The export may be overkill but it makes sure we don't mess us the name of the key

Reply:
Nah I think you're write to export it for sure, I trust your judgement here, I just think what you've told me we should record in the code :)